understand-tidb/introduction-of-tidb-architecture.md

```diff
@@ -1,10 +1,157 @@
+
+### SQL Layer Architecture
+
+![sql-layer-architecture](https://user-images.githubusercontent.com/18818196/121646008-6e22a900-cac7-11eb-954a-fbca825855b0.png)
+
+This is a detailed SQL layer architecture graph. You can read it from left to right.
+
+### Protocol Layer
+
+The leftmost is the Protocol Layer of TiDB, this is the interface to interact with Client, currently TiDB only supports MySQL protocol, the related code is in the `server` package.
+
+The purpose of this layer is to manage the client connection, parse MySQL commands and return the execution result. The specific implementation is according to MySQL protocol, you can refer to [MySQL Client/Server Protocol document](https://dev.mysql.com/doc/internals/en/client-server-protocol.html). If you need to use MySQL protocol parsing and processing functions in your project, you can refer to this module.
+
+The logic for connection establishment is in the `Run()` method of `server.go`, mainly in the following two lines.
+
+```go
+conn, err := s.listener.Accept()
+clientConn := s.newConn(conn)
+go s.onConn(clientConn)
+```
+
+The entry method for a single session processing command is to call the `dispatch` method of the `clientConn` class, where the protocol is parsed and passed to a different handler.
+
+### SQL Layer
+
+Generally speaking, a SQL statement needs to go through a series of processes: 
+
+1. syntax parsing
+2. validity verification
+3. making query plan 
+4. optimizing query plan
+5. generating querier according to plan
+6. executing and returning results
+
+These processes locate at the following modules:
+
+| Package                     | Usage                                                                       |
+| :-------------------------- | :-------------------------------------------------------------------------- |
+| pingcap/tidb/sever          | Interface between protocol layer and SQL layer                              |
+| pingcap/tidb/pingcap/parser | SQL parsing and syntax analyze                                              |
+| pingcap/tidb/planner        | Validation, query plan building, query plan optimizing                      |
+| pingcap/tidb/executor       | Executor generation and execution                                           |
+| pingcap/tidb/distsql        | Send request to TiKV and aggregate return results from TiKV via TiKV Client |
+| pingcap/tidb/store/tikv     | TiKV Client                                                                 |
+
+### KV API Layer
+
+TiDB relies on the underlying storage engine to store and load data. It does not rely on a specific storage engine (such as TiKV), but has some requirements for the storage engine, and any engine that meets these requirements can be used (TiKV is the most suitable one).
```

Comment:
BTW, this is not true, see https://github.com/pingcap/tidb/issues/19111#issuecomment-833394226
Unfortunately, TiDB is highly coupled with TiKV. Should we remove this promise?

Reply:
no, we should keep this promise to make TiDB able to run on different storage layers.

Reply:
Thus I'd like to keep the description as is. You can create a topic on the internals forum to discuss the compatibility further.